src/apple/cli.rs

```diff
@@ -377,13 +377,18 @@ impl Exec for Input {
                 let mut host_env = HashMap::<&str, &OsStr>::new();
 
                 // Host flags that are used by build scripts
-                let macos_isysroot = {
+                let (macos_isysroot, library_path) = {
                     let macos_sdk_root =
                         sdk_root.join("../../../../MacOSX.platform/Developer/SDKs/MacOSX.sdk");
                     if !macos_sdk_root.is_dir() {
                         return Err(Error::MacosSdkRootInvalid { macos_sdk_root });
                     }
-                    format!("-isysroot {}", macos_sdk_root.display())
+                    (
+                        format!("-isysroot {}", macos_sdk_root.display()),
+                        // We need to manually specify a search path to link against the host's libraries for MacOs Big Sur
```

Comment:
More direct description of why we have this:
```suggestion
                        // Prevents linker errors in build scripts and proc macros:
```
Also, this should be on the `LIBRARY_PATH` insert instead of here.